Sources/App/configure.swift

```diff
@@ -38,7 +38,7 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
 
     services.register(OnTapController.self)
     services.register(KarmaController.self)
-    services.register(APIKeyStorage.self)
+    services.register(try Secrets.detect())
```

Comment:
This just feels odd. I understand what you are trying to do, with only calling detect() once but I feel like there is a better way. Im guessing we can't do it by conforming Secrets to ServiceType right ? As that would call it multiple times

Reply:
Yeah, this is the way to just execute the `detect()` code once. Because `Secrets` is a value type, this instance will be copied when requested. Because of this, setting it up this way is thread safe. The other form of this method is `services.register(try Secrets.detect(), as: Secrets.self)` not sure if that gets you anything. This API is provided to register instances, the method signature is `register<S>(_ instance: S, as interfaces: [Any.Type] = [])`. I really wish they'd include the parameter names for the first parameter as it's super unclear which thing is being called.